docs/langref/relay_expr.rst

```diff
@@ -0,0 +1,396 @@
+====================
+Expressions in Relay
+====================
+
+The Relay IR is a pure, expression-oriented language with distinct
+dataflow and control flow language fragments.
+The dataflow fragments of a program (i.e., those without
+calls to recursive functions or branching) can be
+viewed as a traditional computation graph when writing and
+expressing transformations.
+
+The below sections describe the different expressions in Relay
+as well as give details of their semantics.
+
+Variables
+=========
+
+Relay allows for local and global variables. Our design is based on
+that of LLVM, which differentiates between identifier types; a writer of
+optimizations can thus determine what a variable references
+simply by knowing the kind of identifier.
+
+Global variables are written with `@`, local variables are written
+with `%`, and variables written without a sigil correspond to operator
+names.
+
+The distinction between global and local identifiers
+makes certain kinds of transformation easier. For example,
+inlining a global definition requires no analysis: simply inlining
+the definition suffices.
+
+Global Variable
+~~~~~~~~~~~~~~~~~~
+
+Global identifiers are prefixed by the `@` sigil, such as "`@global`".
+A global identifier always references a globally visibly definition contained in the environment.
+Global identifiers must be unique.
+
+See :py:class:`~tvm.relay.expr.GlobalVar` for its implementation
+and documentation.
+
+Local Variable
+~~~~~~~~~~~~~~~~~
+
+Local identifiers are prefixed by the :code:`%` sigil,
+such as ":code:`%local`". A local identifier always references
+a function argument or a variable bound in a :code:`let` expression.
+A local variable will be scoped to the function where it
+appears or the :code:`let` expression where it is bound, respectively.
+
+Suppose the local variable :code:`%a` has been defined in a scope
```

Comment:
I think an annotated text format example would be good here.

Reply:
I added an example, which I hope illustrates the point